### ISSUE TITLE
[FR] Interfacing with closed loop stepper

### ISSUE THREAD

**Issue by alexxy** [MarlinFirmware/Marlin#12496]
What is the best way to interface with closed loop stepper motor?

Currently I working on simple design for closed loop stepper comntroller that can be very cheap but will its work. So basicaly it is stm32 mcu with tmc motor driver and magnetic encoder. So the main question is interface between control board and comntroller. Ideas?

**Comment by comps** [comment]
See #11772 and the discussion in it.

**Comment by alexxy** [comment]
Good point. But idea is to make motor controller board to all of position control thing.

**Comment by paukstelis** [comment]
There have been some discussion.
https://github.com/MarlinFirmware/Marlin/issues/11113

**Comment by boelle** [comment]
@alexxy 

This Issue Queue is for Marlin bug reports and development-related issues, and we prefer not to handle user-support questions here. (As noted on [this page](https://github.com/MarlinFirmware/Marlin/blob/1.1.x/.github/contributing.md#i-dont-want-to-read-this-whole-thing-i-just-have-a-question).) For best results getting help with configuration and troubleshooting, please use the following resources:

- RepRap.org [Marlin Forum](http://forums.reprap.org/list.php?415)
- [Tom's 3D Forums](https://discuss.toms3d.org/)
- Facebook Group ["Marlin Firmware"](https://www.facebook.com/groups/1049718498464482/)
- Facebook Group ["Marlin Firmware for 3D Printers"](https://www.facebook.com/groups/3Dtechtalk/)
- [Marlin Configuration](https://www.youtube.com/results?search_query=marlin+configuration) on YouTube
- Marlin Discord server. Join link: https://discord.gg/n5NJ59y

After seeking help from the community, if the consensus points to to a bug in Marlin, then you should post a bug report.
Before posting a bug report please test with `bugfix-2.0.x` to check if problem is gone

**Comment by boelle** [comment]
@thinkyhead i think we can close this one

**Comment by alexxy** [comment]
Its not user support question. I'm developing board similar to nano zero or machaduino. So I rise question about interfacing with Marlin.

So possibilityes are:

- step/dir/enable interface + additional pin to report that movement done
- step/dir/enable interface + spi or i2c
- can bus to rule em all
- i2c to controll everything
- spi to control everything


Or any other suggestions? I think in case of using closed-loop drives marlin need some kind of feedback to  know if movement finished or not, or it get an error.

I tested encoder resolution and stability: its ~0.02 degree (for example for 200 steps/rev stepper with 16 microstep we need ~0.1125 degree, so resolution is enough for 64-128 microsteps... )

**Comment by alexxy** [comment]
I now implementing interface in  that way
1. closed-loop board exposes step, dir, enable pins that are translated into step dir enable of driver
2. closed-loop board exposes additional two pins: ready and error that are used for reporting status
3. closed-loop board exposes spi for driver configuration (as well as uart)

**Comment by crazyp1g** [comment]
@alexxy   
Why not consider like that TMC 4361 and TMC5160

**Comment by boelle** [comment]
@thinkyhead Another one that needs to be closed. Sorry, I got bored.

**Comment by InsanityAutomation** [comment]
No, there is still plenty to be done around closed loop steppers. I just got a couple in 2 days ago to start playing with. The TMC motion controllers above would be hugely beneficial as well.
